src/lib/services/web-decoder.ts

```
@@ -0,0 +1,92 @@
+    setLastWebDecoderFailure();
+    return Promise.reject(err);
+  }
+
+  try {
+    await sock.open();
+    console.log('opened', sock);
+    const requests = [];
+
+    events.forEach((event: HistoryEventWithId) => {
+      let payloads: Payload[] = [];
+
+      // Probably want to elaborate on the types for this;
+      switch (event.attributes.type) {
+        case 'activityTaskCompletedEventAttributes':
+          payloads = event.attributes.result.payloads;
+          break;
+        case 'signalExternalWorkflowExecutionInitiatedEventAttributes':
+        case 'workflowExecutionStartedEventAttributes':
+        case 'activityTaskScheduledEventAttributes':
+          payloads = event.attributes.input.payloads;
+          break;
+      }
+
+      console.log({ payloads });
+
+      payloads.forEach((payload, i) => {
+        requests.push(
+          sock
+            .sendRequest({ payload: JSON.stringify(payload) })
+            .then((response) => {
+              let currentPayload = null;
+              try {
+                currentPayload = JSON.parse(response.content);
+                setLastWebDecoderSuccess();
+              } catch {
+                // This doesn't seem to be a failure the worker _could_ send back a text response
+                currentPayload = response.content;
+                setLastWebDecoderFailure();
+              }
+              switch (event.attributes.type) {
+                case 'activityTaskCompletedEventAttributes':
+                  event.attributes.result.payloads = currentPayload;
+                  break;
+                case 'signalExternalWorkflowExecutionInitiatedEventAttributes':
+                case 'workflowExecutionStartedEventAttributes':
+                case 'activityTaskScheduledEventAttributes':
+                  event.attributes.input.payloads = currentPayload;
+                  break;
+              }
+            }),
+        );
+      });
+    });
+    await Promise.all(requests);
```

Do you want to return this? Otherwise any errors will be swallowed.